common/src/vote/submission.rs

```diff
@@ -298,14 +298,25 @@ impl<T: Serialize> Serialize for EpochVoteSubmissions<T> {
     where
         S: Serializer,
     {
-        let inner = (
-            &self.total_submission_weight,
-            &self.most_voted_key,
-            &self.submitters,
-            &self.submission_weights,
-            &self.submissions,
-        );
-        serde::Serialize::serialize(&inner, serde_tuple::Serializer(serializer))
+        if let Some(key) = &self.most_voted_key {
+            let inner = (
+                &self.total_submission_weight,
+                &key,
```

Comment:
Just one note, this may be serialized as `[u8]` instead of a slice of bytes in CBOR, but I am not sure. In previous cases I had to mark the type as serde `with=serde_bytes` so it used the right type expected by Go, but I am not sure. I guess we'll realize once we test it.